From 63a71cc2af85e535e63ef047fc49bfbfdc39ce9b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Wed, 20 Jun 2018 16:04:55 +0300
Subject: [PATCH 313/706] arm64: dts: freescale: ls1028a: add crypto node
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

[TODO: validate with Geoff W. that v5.0 is allocated to this device.]

LS1028A has a SEC v5.0 compatible security engine.

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
(cherry picked from commit c77398fb65ece3ab52e7dc76ba130046bf517d9c)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-cfp.dts    |  1 +
 .../boot/dts/freescale/fsl-ls1028a-pxp.dts    |  1 +
 .../boot/dts/freescale/fsl-ls1028a-simu.dts   |  1 +
 .../arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 39 +++++++++++++++++++
 4 files changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-cfp.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-cfp.dts
index 1b3536cc584c..604a11c639c2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-cfp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-cfp.dts
@@ -53,6 +53,7 @@
 	compatible = "fsl,ls1028a-simu", "fsl,ls1028a";
 
 	aliases {
+		crypto = &crypto;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
index 8cf5bf254af6..5e01cfa48a0e 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-pxp.dts
@@ -53,6 +53,7 @@
 	compatible = "fsl,ls1028a-simu", "fsl,ls1028a";
 
 	aliases {
+		crypto = &crypto;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts
index e87ecacad106..10459694c0b6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-simu.dts
@@ -53,6 +53,7 @@
 	compatible = "fsl,ls1028a-simu", "fsl,ls1028a";
 
 	aliases {
+		crypto = &crypto;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index 69e6d6b43dc6..ebc69ecfc9c6 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -250,6 +250,45 @@
 				     <0 208 4>, <0 209 4>;
 		};
 
+		crypto: crypto@8000000 {
+			compatible = "fsl,sec-v5.0", "fsl,sec-v4.0";
+			fsl,sec-era = <10>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0x00 0x8000000 0x100000>;
+			reg = <0x00 0x8000000 0x0 0x100000>;
+			interrupts = <GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>;
+			dma-coherent;
+
+			sec_jr0: jr@10000 {
+				compatible = "fsl,sec-v5.0-job-ring",
+					     "fsl,sec-v4.0-job-ring";
+				reg	= <0x10000 0x10000>;
+				interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
+			sec_jr1: jr@20000 {
+				compatible = "fsl,sec-v5.0-job-ring",
+					     "fsl,sec-v4.0-job-ring";
+				reg	= <0x20000 0x10000>;
+				interrupts = <GIC_SPI 141 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
+			sec_jr2: jr@30000 {
+				compatible = "fsl,sec-v5.0-job-ring",
+					     "fsl,sec-v4.0-job-ring";
+				reg	= <0x30000 0x10000>;
+				interrupts = <GIC_SPI 142 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
+			sec_jr3: jr@40000 {
+				compatible = "fsl,sec-v5.0-job-ring",
+					     "fsl,sec-v4.0-job-ring";
+				reg	= <0x40000 0x10000>;
+				interrupts = <GIC_SPI 143 IRQ_TYPE_LEVEL_HIGH>;
+			};
+		};
+
 		ddr: memory-controller@1080000 {
 			compatible = "fsl,qoriq-memory-controller";
 			reg = <0x0 0x1080000 0x0 0x1000>;
-- 
2.17.1

